### PR TITLE
feat: get dataset file upload URL api

### PR DESCRIPTION
**Pull request by bhudnell**
Issue #, if available:

Description of changes:
- Added API route `GET /datasets/:datasetId/upload-requests?filenames=file1,file2` to get presigned URLs for file upload to the dataset

Checklist:

* [x] Have you successfully deployed to an AWS account with your changes?
* [x] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: solutions/swb-reference/src/services/dataSetService.ts [hunk 1]
```diff
@@ -34,6 +34,8 @@ import { SwbAuthZSubject } from '../constants';
 import { getProjectAdminRole, getResearcherRole } from '../utils/roleUtils';
 import { Associable, DatabaseServicePlugin } from './databaseService';
 
+const timeToLiveSeconds: number = 60; // 1 min
```
Comment (DaveMadsen) [inline solutions/swb-reference/src/services/dataSetService.ts:37]: this seems a little to strict. I'm thinking `60 * 5` by be a better TTL or even as much as `60 * 15`. What you have offered here is 1 minute for a user to take an array of URLs an act on all of them before they expire.

**Review by DaveMadsen — CHANGES_REQUESTED** [review]
I think your presigned URLs need a bit more than 1 minute TTL. 5 minutes? 15 minutes? (No more than 15 minutes IMO)

File: solutions/swb-reference/src/services/dataSetService.ts (hunk 1 above)
Comment (bhudnell) [reply solutions/swb-reference/src/services/dataSetService.ts:37]: Will wait for SWB guidance on this one

Comment (kpark277) [reply solutions/swb-reference/src/services/dataSetService.ts:37]: 15 minutes sounds reasonable.

**Review by DaveMadsen — APPROVED** [review]

File: solutions/swb-reference/src/services/dataSetService.ts (hunk 1 above)
Comment (bhudnell) [reply solutions/swb-reference/src/services/dataSetService.ts]: updated.

**Review by kpark277 — APPROVED** [review]

**Pull request merged**